components/gfx/paint_thread.rs

```diff
@@ -373,7 +375,7 @@ impl<C> PaintThread<C> where C: PaintListener + Send + 'static {
     pub fn create(id: PipelineId,
                   url: Url,
                   chrome_to_paint_chan: Sender<ChromeToPaintMsg>,
-                  layout_to_paint_port: Receiver<LayoutToPaintMsg>,
+                  layout_to_paint_port: IpcReceiver<LayoutToPaintMsg>,
```

Comment:
This is replacing MPSC channels with IPC channels for the single-process case (still the default); it was (obviously) already using IPC channels in the multi-process case.

I don't think we should rely on single-process optimisation if we ever want to make the multi-process case default and/or actually viable. So I don't think this change is bad per se -- it just brings out problems that would have come up anyway if the tests were run with `-m`...